lib/internal/quic/core.js

```diff
@@ -727,15 +727,18 @@ class QuicSocket extends EventEmitter {
   }
 
   // Bind the UDP socket on demand, only if it hasn't already been bound.
-  // Function is a non-op if the socket is already bound
+  // Function is a non-op if the socket is already bound or in the process of
+  // being bound, and will call the callback once the socket is ready.
   [kMaybeBind](callback = () => {}) {
+    if (this.#state !== kSocketUnbound) {
+      if (this.bound)
+        return process.nextTick(callback);
+      else
+        return this.on('ready', callback);
+    }
     // This socket will be in a pending state until it is bound. Once bound,
     // the this[kReady]() method will be called, switching the state to
-    // kSocketBound and notifying the associated sessions
-    // TODO(@jasnell): If the socket is already bound, the callback should
-    // be invoked with an error.
```

Comment:
@jasnell I assume this TODO didn’t exist without a reason … do you remember more about this?

Reply:
ummmm... I think this is no longer relevant, particularly with the changes you've been making here.